webui/react/src/hooks/useRestApi.ts

```diff
@@ -45,8 +45,11 @@ type Output<T> = [
 
 const defaultReducer = <T>(state: State<T>, action: Action<T>): State<T> => {
   switch (action.type) {
-    case ActionType.SetData:
-      return { ...state, data: action.value, hasLoaded: true, isLoading: false };
+    case ActionType.SetData: {
+      // TODO unnecessary?
```

Comment:
this isn't necessary for this PR but @hkang1 do you think this is necessary in general or react is doing something similar under the hood?

Reply:
I wasn't sure either, so I looked into it. Looks like react uses the native JS `Object.is` function to [compare objects](https://reactjs.org/docs/hooks-reference.html#bailing-out-of-a-dispatch). According to the [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Description) it only compares to see if the references are the same for the objects. So in this example, `state.data` is always going to be a different reference than `action.value` (unless you are actually passing in the previous value of `action.value`). But looking at this, it looks like are want to check to make sure the shape and content of the objects are identical so this is NOT redundant code.

Reply:
TL;DR This is great to have :)